crates/nu-command/src/conversions/split_cell_path.rs

```diff
@@ -0,0 +1,168 @@
+                example: "$.5?.c | split cell-path",
+                result: Some(Value::test_list(vec![
+                    Value::test_record(record! {
+                        "value" => Value::test_int(5),
+                        "optional" => Value::test_bool(true),
+                    }),
+                    Value::test_record(record! {
+                        "value" => Value::test_string("c"),
+                        "optional" => Value::test_bool(false),
+                    }),
+                ])),
+            },
+            Example {
+                description: "Split a complex cell-path",
+                example: r#"$.a.b?.1."2"."c.d" | split cell-path"#,
+                result: Some(Value::test_list(vec![
+                    Value::test_record(record! {
+                        "value" => Value::test_string("a"),
+                        "optional" => Value::test_bool(false),
+                    }),
+                    Value::test_record(record! {
+                        "value" => Value::test_string("b"),
+                        "optional" => Value::test_bool(true),
+                    }),
+                    Value::test_record(record! {
+                        "value" => Value::test_int(1),
+                        "optional" => Value::test_bool(false),
+                    }),
+                    Value::test_record(record! {
+                        "value" => Value::test_string("2"),
+                        "optional" => Value::test_bool(false),
+                    }),
+                    Value::test_record(record! {
+                        "value" => Value::test_string("c.d"),
+                        "optional" => Value::test_bool(false),
+                    }),
+                ])),
+            },
+        ]
+    }
+}
+
+fn split_cell_path(val: CellPath, span: Span) -> Result<Value, ShellError> {
+    #[derive(IntoValue)]
+    struct PathMemberRecord {
+        value: Value,
+        optional: bool,
+    }
+
+    impl PathMemberRecord {
+        fn from_path_member(pm: &PathMember) -> Self {
+            let value = match pm {
+                PathMember::String { val, span, .. } => Value::String {
+                    val: val.clone(),
+                    internal_span: *span,
+                },
+                PathMember::Int { val, span, .. } => Value::Int {
+                    val: *val as i64,
+                    internal_span: *span,
+                },
+            };
+            let optional = match pm {
+                PathMember::String { optional, .. } | PathMember::Int { optional, .. } => *optional,
+            };
+            Self { value, optional }
+        }
+    }
+
+    let members = val
+        .members
+        .iter()
+        .map(|pm| {
+            PathMemberRecord::from_path_member(pm).into_value(match pm {
+                PathMember::String { span, .. } | PathMember::Int { span, .. } => *span,
+            })
+        })
```

Comment:
Not really performance critical code as things stand right now but for good practice:

This can consume the `members` and reuse the `String` allocations based on the outside signature.

Reply:
Thanks for the recommendation!